ubiconfig/_impl/loaders/gitlab.py

```diff
@@ -23,80 +23,101 @@ def __init__(self, url):
         self._url = url
         self._session = requests.Session()
         self._repo_api = RepoApi(self._url.rstrip("/"))
+        self._branches = self._get_branches()
         self._files_branch_map = self._pre_load()
 
-    def load(self, file_name):
+    def load(self, file_name, version=None):
         """ Load file from remote repository.
         :param file_name: filename that is on remote repository in any branch
         """
-        # find the right branch from the mapping
-        branch_sha1 = self._files_branch_map[file_name]
-        version = branch_sha1[0][3:]  # such as ubi7, ubi7.1
+        if file_name not in self._files_branch_map:
+            raise ValueError("Couldn't find file %s in remote repo" % file_name)
 
-        config_file_url = self._repo_api.get_file_content_api(file_name, branch_sha1[1])
-        LOG.info("Loading configuration file from remote: %s", file_name)
+        sha1 = self._branches.get(version)
+        if version and not sha1:
+            LOG.warning(
+                "Didn't find version %s, will try to find %s in default",
```

Comment:
Couldn't find... sounds better. Also, where did not find it? Would be nice to include.

Reply:
makes sense.